iscript/src/iscript/script.py

```diff
@@ -8,52 +8,79 @@
 
 from iscript.exceptions import IScriptError
 from iscript.mac import notarize_1_behavior, notarize_3_behavior, notarize_behavior, sign_and_pkg_behavior, sign_behavior, single_file_behavior
-from iscript.macvpn import notarize_vpn_behavior
+from iscript.macvpn import vpn_behavior
 from iscript.util import get_sign_config
 
 log = logging.getLogger(__name__)
 
 
+def check_dep_behavior(task, behavior, supported_behaviors):
+    """Check behavior for dep signing.
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+
+    Raises:
+        IScriptError if behavior is unsupported
+
+    Returns:
+        str: the behavior
+    """
+    if behavior == "mac_notarize" and "mac_sign_and_pkg" in supported_behaviors:
+        behavior = "mac_sign_and_pkg"
+    if behavior == "mac_notarize_vpn" and "mac_sign_and_pkg_vpn" in supported_behaviors:
+        behavior = "mac_sign_and_pkg_vpn"
+
+    if behavior not in supported_behaviors:
+        raise IScriptError("Unsupported behavior {} given scopes {}!".format(behavior, task["scopes"]))
+    return behavior
+
+
+def get_behavior_function(behavior):
+    """Map a behavior to a function.
+
+    Args:
+        behavior (str): signing behavior
+
+    Returns:
+        tuple: (func, {args})
+    """
+    functions = {
+        "mac_geckodriver": (single_file_behavior, {}),
+        "mac_single_file": (single_file_behavior, {}),
+        "mac_notarize": (notarize_behavior, {}),
+        "mac_notarize_vpn": (vpn_behavior, {"notarize": True}),
+        "mac_sign_and_pkg_vpn": (vpn_behavior, {"notarize": False}),
+        "mac_notarize_part_1": (notarize_1_behavior, {}),
+        "mac_notarize_part_3": (notarize_3_behavior, {}),
+        "mac_sign": (sign_behavior, {}),
+        # For staging releases; or should we mac_notarize but skip notarization
+        # for dep?
+        "mac_sign_and_pkg": (sign_and_pkg_behavior, {}),
+    }
+    if behavior not in functions:
+        raise IScriptError("iscript behavior {} not implemented!".format(behavior))
+    return functions[behavior]
+
+
 async def async_main(config, task):
     """Sign all the things.
 
     Args:
         config (dict): the running config.
         task (dict): the running task.
-
```

Comment:
Historically, docstring linting has said we have to end with an empty line, but this is ok if that rule has changed.